foodnet/membership/models.py

```diff
@@ -53,6 +53,18 @@ def is_complete(self):
             return True
         return False
 
+    def has_admin_permission(self, department):
+        return self.departmentadministrator_set.filter(department=department).exists()
+
+    def can_be_edited_by(self, user_profile):
+        # can be edited by user_profile if the user_profile is an admin of
+        # the self.member.department department
+        departments = self.member.department_set.all()
+        for department in departments.iterator():
+            if user_profile.has_admin_permission(department=department):
```

Comment:
It may be a premature optimization but if we imagine we want to display 50 members each having at least 2 users on admin page the method will execute 100 sql queries and for each query iterate over some subset of departments.
Do we really want to do it this way(?) Can you think about some other way of doing this(?)